src/index/MetaDataHandler.h

```diff
@@ -0,0 +1,302 @@
+    // And put this empty metaData to the FullRelationMetaData class.
+    FullRelationMetaData emptyMetaData = FullRelationMetaData(-1, -1, -1, 1.0, 1.0, false, false);
+  };
+
+}
+
+// _____________________________________________________________________
+template <class M>
+class MetaDataWrapperMmap {
+ public:
+  using Iterator = ExtVecWrapperImpl::Iterator<M>;
+
+  // _________________________________________________________
+  MetaDataWrapperMmap() = default;
+
+  // _________________________________________________________________
+  void setup(size_t numEls, const string& filename) {
+    _vec.setupNewFile(numEls, emptyMetaData, filename);
+  }
+
+  // __________________________________________________________________
+  // TODO: mark that this is the read version, e.g by using
+  // an enum-Flag
+  void setup(const string& filename) {
+    _vec.setupExistingFile(filename);
+    // get the size by checking for non-empty entries
+    // TODO: this is not really efficient.
+    // Currently we also  have to loop through all entries to find the blocks
+    // (and  normally there are no blocks with  the dense metaData)
+    // maybe save this in the mmap file by templating it with additional meta
+    // data to be written at the  end
+    for (const auto& el : _vec) {
+      _size += (el != emptyMetaData);
+    }
+  }
+  
+  // ___________________________________________________________
+  size_t size() const {return _size;}
+
+  // __________________________________________________________________
+  const Iterator cbegin() const {
+    Iterator it(0, _vec.begin(), &_vec);
+    it.goToNexValidEntry();
+    return it;
+  }
+
+  // __________________________________________________________________________
+  const Iterator cend() const {
+    return Iterator(_vec.size(), _vec.end(), &_vec);
+  }
+
+
+  // ____________________________________________________________
+  void set(Id id, const FullRelationMetaData& value) {
+    // TODO: Exception or assertion? this should  never happen
+    AD_CHECK(id < _vec.size());
+    bool previouslyEmpty = _vec[id] == emptyMetaData;
+
+    // check that we never insert the empty key
+    AD_CHECK(value != emptyMetaData);
+    //TODO: assert, that this should never be true
+    _vec[id] = value;
+    if (previouslyEmpty) { _size++;}
+  }
+
+  // __________________________________________________________
+  const FullRelationMetaData& getAsserted(Id id) const {
+    const auto& res = _vec[id];
+    AD_CHECK(res != emptyMetaData);
+    return res;
+  }
+
+  // ________________________________________________________
+  size_t count(Id id) const {
+    // can either be 1 or 0 for map-like types
+    return _vec[id] != emptyMetaData;
+  }
+ 
+ private:
+  // the empty key, must be the first member to be initialized
+  FullRelationMetaData emptyMetaData = FullRelationMetaData(-1, -1, -1, 1.0, 1.0, false, false);
+  size_t _size = 0;
+  M _vec;
+};
+
+// TODO: this is basically the same like above. Can we
```

Comment:
If the other one works well I'd just remove it, though you could keep it for benchmarks